tailcfg/tailcfg_view.go

```diff
@@ -2,99 +2,323 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Code generated by tailscale.com/util/codegen (viewer); DO NOT EDIT.
+
 package tailcfg
 
 import (
 	"encoding/json"
 	"errors"
+	"time"
 
+	"inet.af/netaddr"
+	"tailscale.com/types/dnstype"
+	"tailscale.com/types/key"
 	"tailscale.com/types/opt"
+	"tailscale.com/types/structs"
 	"tailscale.com/types/views"
 )
 
-// View returns a read-only accessor for hi.
-func (hi *Hostinfo) View() HostinfoView { return HostinfoView{hi} }
+//go:generate go run tailscale.com/cmd/cloner  -clonefunc=true -type=User,Node,Hostinfo,NetInfo,Login,DNSConfig,RegisterResponse,DERPRegion,DERPMap,DERPNode
```

Comment:
reminder to make the "go generate" step on re-parse package and if type checking or whatever fails then say loudly to stderr, "hey, you gotta run this a second time"